rabbitmq/src/messages.rs

```diff
@@ -73,3 +76,33 @@ impl ResultMessage {
         }
     }
 }
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum WorkerStatusType {
+    Lifecycle,
+    Job,
+    Heartbeat,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum WorkerStatus {
+    Online,
+    Offline,
+}
+impl WorkerStatus {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            WorkerStatus::Online => "online",
+            WorkerStatus::Offline => "offline",
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct StatusMessage {
+    pub worker_name: String,
+    pub status: Option<WorkerStatus>,
+    pub status_type: WorkerStatusType,
+    pub timestamp: DateTime<Utc>,
+    pub job_id: Option<Uuid>,
```

Comment:
Maybe put `job_id` and `status` into `status_type` type? That way we'll get rid of these awkward options. Something like this:

```
pub enum WorkerStatus {
    Lifecycle(WorkerStatus),
    Job(Option<Uuid>),
    Heartbeat,
}
[...]
pub struct StatusMessage {
    pub worker_name: String,
    pub status: WorkerStatus,
    pub timestamp: DateTime<Utc>,
}
```

That way:
1. We make it impossible to represent invalid states (for example, status type of Lifecycle, but status being None instead of Some(Online/Offline))
2. We get rid of the double-meaning of None `job_id`, which for `Job` status type means that we've finished processing a job, while for other status types it means that it's not applicable to given status type.